### PR TITLE
Add pre commit/linters

### PR DESCRIPTION
**Pull request by marcelotrevisani**

### CODE REVIEW

File: .github/workflows/linters.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Linters
```
Comment (nicoddemus) [inline .github/workflows/linters.yml:1]: Instead of setting up a specific action, we can just instead enable pre-commit.ci in this repository... it not only checks that pre-commit runs, but also updates the PR with fixes when possible.

We use all over pytest-dev and in several other projects, cannot recommend enough.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    -   id: check-yaml
+        exclude: tests
+    -   id: check-toml
+    -   id: check-json
+    -   id: check-merge-conflict
+    -   id: pretty-format-json
+        args: [--autofix]
+    -   id: debug-statements
+        language_version: python3
+-   repo: https://github.com/pre-commit/mirrors-isort
```
Comment (nicoddemus) [inline .pre-commit-config.yaml:29]: I strongly prefer https://github.com/asottile/reorder_python_imports, for the reasons cited there: https://github.com/asottile/reorder_python_imports#why-this-style

Having said that, I would then change the order things execute:

- autoflake
- reoder-imports
- black
- flake8

The rationale being: cleanups, formatters, flake checking

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+exclude: (?x)(\.txt)
+repos:
+-   repo: https://github.com/psf/black
```
Comment (nicoddemus) [inline .pre-commit-config.yaml:3]:
```suggestion
-   repo: https://github.com/myint/autoflake
    rev: v1.5.3
    hooks:
    -   id: autoflake
        name: autoflake
        args: ["--in-place", "--remove-unused-variables", "--remove-all-unused-imports"]
        language: python
        files: \.py$
-   repo: https://github.com/psf/black
```

Auto flake is great for removing unused imports and variables. 😁 

(Note the diff adds `autoflake` before black, might be easy to miss)

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -10,30 +10,28 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python: ["3.6", "3.7", "3.8", "3.9", "3.10"]
-        os: [ubuntu-latest, windows-latest]
+        python: ["3.7", "3.8", "3.9", "3.10"]
```
Comment (nicoddemus) [inline .github/workflows/main.yml:13]: Dropping Python 3.6 should be a separate PR I think.

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Thanks for raising this topic @marcelotrevisani!

Left a few comments, let me know what you think.

File: .github/workflows/main.yml (hunk 4 above)
Comment (marcelotrevisani) [reply .github/workflows/main.yml:13]: can we do this together? because setuptools_scm will fail for python 3.6, that was the reason to put it in this PR as well :grin:

Comment (nicoddemus) [reply .github/workflows/main.yml:13]: It is a known issue, but here is the fix: https://github.com/ESSS/pytest-regressions/blob/c2cfa9906a3df3474a7a45decc267389be941eb1/setup.py#L15-L18

File: .github/workflows/linters.yml (hunk 1 above)
Comment (marcelotrevisani) [reply .github/workflows/linters.yml:1]: sure thing, let me add it

Comment (nicoddemus) [reply .github/workflows/linters.yml:1]: @gabrielcnr will need to enable [pre-commit.ci](https://pre-commit.ci/) as he is the admin.

Comment (marcelotrevisani) [reply .github/workflows/linters.yml:1]: It seems just @gabrielcnr can add it

File: .pre-commit-config.yaml (hunk 2 above)
Comment (marcelotrevisani) [reply .pre-commit-config.yaml:29]: I am not strongly attached to it, if you prefer this it is fine for me :)

File: .pre-commit-config.yaml (hunk 3 above)
Comment (marcelotrevisani) [reply .pre-commit-config.yaml:3]: cool! gonna add it

File: .github/workflows/main.yml (hunk 4 above)
Comment (marcelotrevisani) [reply .github/workflows/main.yml:13]: https://github.com/gabrielcnr/pytest-datadir/pull/42

Comment (nicoddemus) [reply .github/workflows/main.yml:13]: Merged

File: .pre-commit-config.yaml (hunk 2 above)
Comment (nicoddemus) [reply .pre-commit-config.yaml:37]: Sure lets do it then. Make sure to also remove `.isort.cfg` as it will no longer be needed.

Comment (marcelotrevisani) [reply .pre-commit-config.yaml]: done!

File: tests/test_pathlib.py [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import sys
+
 import pytest
 
-@pytest.mark.skipif(sys.version_info[0] == 3, reason='Python 2 only')
+
+@pytest.mark.skipif(sys.version_info[0] == 3, reason="Python 2 only")
```
Comment (nicoddemus) [inline tests/test_pathlib.py:6]: This test can be removed. 👍

Comment (marcelotrevisani) [reply tests/test_pathlib.py:6]: done

**Review by nicoddemus — APPROVED** [review]
